akka-cluster-sharding/src/main/resources/reference.conf

```diff
@@ -99,12 +99,36 @@ akka.cluster.sharding {
     # the region with least shards must be greater than (>) the `rebalanceThreshold`
     # for the rebalance to occur.
     # 1 gives the best distribution and therefore typically the best choice.
-    # Increasing the threshold can result in quicker rebalance but has the
-    # drawback of increased difference between number of shards (and therefore load)
-    # on different nodes before rebalance will occur.
     rebalance-threshold = 1
 
-    # The number of ongoing rebalancing processes is limited to this number.
+    # The maximum number of shards to start rebalancing on each rebalance decision.
+    # Increasing this will increase the speed at which rebalancing occurs.
+    # This differs from max-simultaneous-rebalance, in that the latter applies across
+    # multiple rebalance intervals, whereas this only applies to a single rebalance
+    # interval decision.
+    # The smaller of this number and the limit decided by the rebalance-factor will
+    # be used. If this is zero, then just the rebalance-factor will be used. If both
+    # this and the rebalance factor are zero, then the rebalance-threshold will be
+    # used as the limit.
+    rebalance-number = 0
+
+    # The factor to apply to the number of shards in the region with the most shards,
+    # used to decide how many shards to start rebalancing on eac rebalance decision.
+    # Using a factor may be useful because it sets a percentage of total work to
+    # migrate, so when a nodes number of shards is low, it can migrate just a small
+    # number assuming each shard represents a large amount of work, but when it's high,
+    # it can migrate a large number assuming each shard represents a small amount of
+    # work.
+    # Must be a value between 0 and 0.5 (values greater than 0.5 don't make sense since
+    # migrating more than half the shards from one region to another will always
+    # overshoot a balanced distribution).
+    # If this is zero, then the number of shards to migrate is not selected based on
+    # the current number of shards, but rather rebalance-number is used to set an
+    # absolute limit.
+    rebalance-factor = 0.0
```

Comment:
I wonder if this should be a factor of the difference, rather than a factor of the total number of shards. But this works well for the case of adding a new node.

Reply:
Ah i see this is handled by the evenRebalance being another max value